pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md

```diff
@@ -0,0 +1,214 @@
+
+// create default parameters for the SVG layout
+LayoutParameters layoutParameters = new LayoutParameters();
+
+// display line ID instead of line name
+boolean usename = false;
+
+// prefix used for the IDs of all diagram components
+String prefix = "";
+```
+
+## Diagrams from a test network
+We first load the node/breaker test `Network` we are interested in:
+```java
+Network network = FictitiousSwitchFactory.create();
+```
+
+### Generating a voltage level diagram
+We generate a `VoltageLevelDiagram` for voltage level N and then the corresponding SVG diagram file:
+```java
+// create diagram for the voltage level N
+VoltageLevelDiagram voltageLevelDiagram = VoltageLevelDiagram.build(new NetworkGraphBuilder(network), "N", voltageLevelLayoutFactory, usename);
+
+// generate SVG
+voltageLevelDiagram.writeSvg(prefix
+    new DefaultSVGWriter(componentLibrary, layoutParameters),
+    new DefaultDiagramLabelProvider(network, componentLibrary, layoutParameters),
+    new NominalVoltageDiagramStyleProvider(network),
+    Paths.get("/tmp/n.svg"));
+```
+
+We obtain the following SVG:
+
+![N_voltageLevel](img/single-line-diagram_svg-basics/example_n.svg)
+
+Similarly, we could generate a SVG for voltage level C:
+ 
+```java
+// create diagram for the voltage level C
+VoltageLevelDiagram voltageLevelDiagram = VoltageLevelDiagram.build(new NetworkGraphBuilder(network), "C", voltageLevelLayoutFactory, usename);
+
+// generate SVG
+voltageLevelDiagram.writeSvg(prefix,
+    new DefaultSVGWriter(componentLibrary, layoutParameters),
+    new DefaultDiagramLabelProvider(network, componentLibrary, layoutParameters),
+    new NominalVoltageDiagramStyleProvider(network),
+    Paths.get("/tmp/c.svg"));
+```
+ 
+leading to the following diagram:
+
+![C_voltageLevel](img/single-line-diagram_svg-basics/example_c.svg)
+
+### Generating the substation diagram
+In order to build the diagram for the whole substation, named A, containing both voltage levels displayed previously, we need to build the corresponding `SubstationDiagram`:
+```java
+// create diagram for the substation A
+SubstationDiagram substationDiagram = SubstationDiagram.build(
+    new NetworkGraphBuilder(network), "A", new HorizontalSubstationLayoutFactory(),
+    voltageLevelLayoutFactory, usename);
+
+// generate SVG
+substationDiagram.writeSvg(prefix,
+    new DefaultSVGWriter(componentLibrary, layoutParameters),
+    Paths.get("/tmp/a.svg"),
+    new DefaultDiagramLabelProvider(network, componentLibrary, layoutParameters),
+    new NominalVoltageDiagramStyleProvider(network)
+);
+```
+
+We then obtain the following wider SVG file:
+
+![C_voltageLevel](img/single-line-diagram_svg-basics/example_sub_a.svg)
+
+## Diagrams from a CGMES file
+
+First of all, we need to download sample files from ENTSO-E [here](https://docstore.entsoe.eu/Documents/CIM_documents/Grid_Model_CIM/TestConfigurations_packageCASv2.0.zip)
```

Comment:
I think we have factories to create small microgrid (to be checked). I would prefer to be independent from external resources, specially if they decide to remove the file one day

Reply:
I understand, but what is pointed out here is how to create a SVG file from a CGMES file. If we use a factory to create that grid we're losing that part. It's quite easy to import such a file, and it is not a sld feature, sure... but I think we should document it anyway, don't you?
If we want to be independent from external resources, maybe we could host that CGMES file (144ko), or another one?

Reply:
Note that this file are in the repo https://github.com/powsybl/powsybl-cgmes-conformity-assessments

Reply:
Link to file hosted on our github added

Reply:
"internal" file linked instead of entsoe external link.